src/rgw/rgw_op.cc

```diff
@@ -2575,6 +2575,66 @@ void RGWPutObj::pre_exec()
   rgw_bucket_object_pre_exec(s);
 }
 
+class RGWPutObj_CB : public RGWGetDataCB
+{
+  RGWPutObj *op;
+public:
+  RGWPutObj_CB(RGWPutObj *_op) : op(_op) {}
+  virtual ~RGWPutObj_CB() {}
+
+  int handle_data(bufferlist& bl, off_t bl_ofs, off_t bl_len) {
+    return op->get_data_cb(bl, bl_ofs, bl_len);
+  }
+};
+
+int RGWPutObj::get_data_cb(bufferlist& bl, off_t bl_ofs, off_t bl_len)
+{
+  bufferlist bl_tmp;
+  bl.copy(bl_ofs, bl_len, bl_tmp);
+
+  bl_aux.append(bl_tmp);
+
+  return bl_len;
+}
+
+int RGWPutObj::get_data(string bucket_name, string object_name, off_t fst, off_t lst, bufferlist& bl)
+{
+  RGWObjectCtx& obj_ctx = *static_cast<RGWObjectCtx *>(s->obj_ctx);
+  RGWBucketInfo bucket_info;
+  map<string, bufferlist> src_attrs;
+  RGWPutObj_CB cb(this);
+  int ret = 0;
+
+  ret = store->get_bucket_info(obj_ctx, s->src_tenant_name, bucket_name, bucket_info, NULL, &src_attrs);
```

Comment:
@jmunhoz should call get_bucket_info() much earlier (at get_params()), and user the RGWBucketInfo param from that point. So this method should get the resulting rgw_obj (from tenant + bucket_info + object name).